include/networkit/algebraic/CSRGeneralMatrix.hpp

```diff
@@ -395,25 +354,42 @@ class CSRGeneralMatrix {
      * Sorts the column indices in each row for faster access.
      */
     void sort() {
-#pragma omp parallel for schedule(guided)
-        for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i) {
-            if (rowIdx[i + 1] - rowIdx[i] > 1) {
-                quicksort(rowIdx[i], rowIdx[i + 1] - 1);
+#pragma omp parallel
+        {
+            std::vector<index> permutation(nCols);
+#pragma omp for schedule(guided)
+            for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i) {
+                const index startOfRow = rowIdx[i], endOfRow = rowIdx[i + 1];
+                const count nonZerosInRow = endOfRow - startOfRow;
+                if (nonZerosInRow <= 1
+                    || std::is_sorted(columnIdx.begin() + startOfRow, columnIdx.begin() + endOfRow))
+                    continue;
+
+                permutation.resize(nonZerosInRow);
+                std::iota(permutation.begin(), permutation.end(), index{0});
+                std::sort(permutation.begin(), permutation.end(),
+                          [&columnIdx = columnIdx, startOfRow = startOfRow](index x, index y) {
+                              return columnIdx[startOfRow + x] < columnIdx[startOfRow + y];
+                          });
+
+                Aux::ArrayTools::applyPermutation(columnIdx.begin() + startOfRow,
+                                                  columnIdx.begin() + endOfRow,
+                                                  permutation.begin());
+
+                Aux::ArrayTools::applyPermutation(nonZeros.begin() + startOfRow,
+                                                  nonZeros.begin() + endOfRow, permutation.begin());
             }
         }
 
-#ifndef NETWORKIT_SANITY_CHECKS
+#ifdef NETWORKIT_SANITY_CHECKS
         bool sorted = true;
         tlx::unused(sorted);
 #pragma omp parallel for
-        for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i) {
-            for (index j = rowIdx[i] + 1; j < rowIdx[i + 1]; ++j) {
-                if (columnIdx[j - 1] > columnIdx[j]) {
-                    sorted = false;
-                    break;
-                }
-            }
-        }
+        for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i)
+            if (sorted
+                && !std::is_sorted(columnIdx.begin() + rowIdx[i],
+                                   columnIdx.begin() + rowIdx[i + 1]))
+                sorted = false;
```

Comment:
I would simply `if(sorted) assert(std::is_sorted(...))` here instead of modifying the `sorted` member.

Reply:
Agree, I addressed this.